flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/schema/StreamTableSourceTable.scala

```diff
@@ -39,6 +39,8 @@ class StreamTableSourceTable[T](
     val fieldCnt = fieldNames.length
 
     val rowtime = tableSource match {
+      case emptyStringTimeSource : DefinedRowtimeAttribute if emptyStringTimeSource.getRowtimeAttribute.trim.equals("")  =>
```

Comment:
We should move the check `getRowtimeAttribute == null` (currently `case _ => None`) to the top to avoid an NPE if the attribute is null.